core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java

```diff
@@ -73,54 +82,58 @@ public int getPageSize() {
     return pageSize;
   }
 
-  private static ColumnPage createVarLengthPage(DataType dataType, int pageSize) {
+  private static ColumnPage createVarLengthPage(DataType dataType, int pageSize, int scale,
+      int precision) {
     if (unsafe) {
       try {
-        return new UnsafeVarLengthColumnPage(dataType, pageSize);
+        return new UnsafeVarLengthColumnPage(dataType, pageSize, scale, precision);
       } catch (MemoryException e) {
         throw new RuntimeException(e);
       }
     } else {
-      return new SafeVarLengthColumnPage(dataType, pageSize);
+      return new SafeVarLengthColumnPage(dataType, pageSize, scale, precision);
     }
   }
 
-  private static ColumnPage createFixLengthPage(DataType dataType, int pageSize) {
+  private static ColumnPage createFixLengthPage(DataType dataType, int pageSize, int scale,
+      int precision) {
     if (unsafe) {
       try {
-        return new UnsafeFixLengthColumnPage(dataType, pageSize);
+        return new UnsafeFixLengthColumnPage(dataType, pageSize, scale, precision);
       } catch (MemoryException e) {
         throw new RuntimeException(e);
       }
     } else {
-      return new SafeFixLengthColumnPage(dataType, pageSize);
+      return new SafeFixLengthColumnPage(dataType, pageSize, scale, pageSize);
     }
   }
 
-  private static ColumnPage createPage(DataType dataType, int pageSize) {
+  private static ColumnPage createPage(DataType dataType, int pageSize, int scale, int precision) {
     if (dataType.equals(BYTE_ARRAY) | dataType.equals(DECIMAL)) {
-      return createVarLengthPage(dataType, pageSize);
+      return createVarLengthPage(dataType, pageSize, scale, precision);
     } else {
-      return createFixLengthPage(dataType, pageSize);
+      return createFixLengthPage(dataType, pageSize, scale, precision);
     }
   }
 
-  public static ColumnPage newVarLengthPath(DataType dataType, int pageSize) {
+  public static ColumnPage newVarLengthPath(DataType dataType, int pageSize, int scale,
```

Comment:
can you create another function of `newVarLengthPage` which accept `DataType dataType, int pageSize` only and pass -1, -1 to this function